components/editor/modules/link/ui.js

```diff
@@ -10,7 +15,100 @@ import {
   buttonStyles
 } from '../../utils'
 
-import MetaForm from '../../utils/MetaForm'
+export const usersQuery = gql`
+query users($search: String!) {
+  users(search: $search, role: "editor") {
+    firstName
+    lastName
+    id
+  }
+}
+`
+export class SearchUserForm extends Component {
+  constructor (...args) {
+    super(...args)
+    this.state = {
+      items: [],
+      filter: '',
+      value: null
+    }
+    this.filterChangeHandler = this.filterChangeHandler.bind(this)
+    this.changeHandler = this.changeHandler.bind(this)
+  }
+  componentDidMount () {
+    this._isMounted = true
+  }
+
+  componentWillUnmount () {
+    this._isMounted = false
+  }
+
+  filterChangeHandler (value) {
+    if (this._isMounted) {
+      this.setState(
+        state => ({
+          ...this.state,
+          filter: value
+        }),
+        () => this.loadUsers()
+      )
+    }
+  }
+
+  loadUsers () {
+    this.props.client
+      .query({
+        query: usersQuery,
+        variables: { search: this.state.filter }
+      })
+      .then(
+        ({ data }) => {
+          if (this._isMounted) {
+            this.setState(state => ({
+              ...this.state,
+              items: data.users.slice(0, 5).map(v => ({
+                value: v.id,
+                text: `${v.firstName} ${v.lastName}`
+              }))
+            }))
+          }
+        }
+      )
+      .catch(
+        error => {
+          throw error
+        }
+      )
+  }
+
+  changeHandler (value) {
+    if (this._isMounted) {
+      this.setState(
+        state => ({
+          ...this.state,
+          value: null
+        }),
+        () => this.props.onChange(value)
+      )
+    }
+  }
+
+  render () {
+    const { items, filter, value } = this.state
+    return (
+      <Autocomplete
+        label={this.props.t('metaData/field/authors', undefined, 'Autor suchen')}
+        items={items}
+        filter={filter}
+        value={value}
+        onChange={this.changeHandler}
+        onFilterChange={this.filterChangeHandler}
+        />
+    )
+  }
+}
+
+const ConnectedSearchUserForm = withT(withApollo(SearchUserForm))
```

Comment:
would be nicer to use a stateless sub component for the query:
```js
const AutocompleteWithUsers = graphql(usersQuery, {
  skip: props => !props.filter,
  options: ({ filter }) => ({ variables: { search: filter } }),
  props: ({ data: { users = [] } }) => ({
    items: users.slice(0, 5).map(v => ({
      value: v.id,
      text: `${v.firstName} ${v.lastName}`
    })
  })
})(Autocomplete);
```

And then simply use `<AutocompleteWithUsers {...propsWithoutItems} />` above

(would also eliminate `this._isMounted` and potentially optimize the query)